### PR TITLE
1471 - Add BSC chain

### PR DESCRIPTION
**Pull request by jorgedclopes**
- Add BSC support.
- Add pancakeswap exchange.

This development goes in hand with this [dapp branch](https://github.com/levelkdev/swapr-dapp/tree/feature/1471-BSC-chain).

To test the application this library needs to linked to the dapp. PM me if you need help.

### CODE REVIEW

File: src/entities/currency.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import JSBI from 'jsbi'
 import { ChainId, SolidityType } from '../constants'
 import { validateSolidityTypeInstance } from '../utils'
 
+const BURN_ADDRESS = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'
```
Comment (adamazad) [inline src/entities/currency.ts:6]: There's no burn address, we need a better name :)

**Review by karczuRF — APPROVED** [review]

File: src/entities/trades/uniswap-v2/constants.ts [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+export const ETH: Record<number, Token> = {
+  [ChainId.BSC_MAINNET]: new Token(
+    ChainId.BSC_MAINNET,
+    '0x2170Ed0880ac9A755fd29B2688956BD959F933F8',
+    18,
+    'ETH',
+    'Binance-Peg Ethereum'
+  ),
+}
+
```
Comment (adamazad) [inline src/entities/trades/uniswap-v2/constants.ts:18]: What's the difference between this ETH and WETH? My understanding is Ether is on Ether on Mainnet, and true L2's like Arbitrum and Optimism. However, on sidechains such as Gnosis, Polygon PoS, and BNB, there's an IOU claim represented in WETH.

**Review by adamazad — COMMENTED** [review]
Looks good, just left out two comments :)

File: src/entities/trades/uniswap-v2/constants.ts (hunk 2 above)
Comment (jorgedclopes) [reply src/entities/trades/uniswap-v2/constants.ts:18]: ye... that token is Binance-Peg. My bad. I'll remove it.

File: src/entities/currency.ts (hunk 1 above)
Comment (jorgedclopes) [reply src/entities/currency.ts]: Changed to NULL_ADDRESS

**Review by adamazad — APPROVED** [review]
Good work

**Pull request merged**